File: python/medium/2145_count_the_hidden_sequences.py
class Solution:
    def numberOfArrays(self, differences: List[int], lower: int, upper: int) -> int:
        current = 0
        min_val, max_val = 0, 0
        for diff in differences:
            current += diff
            min_val = min(min_val, current)
            max_val = max(max_val, current)
        possible_starts = upper - lower + 1 - (max_val - min_val)
        return max(0, possible_starts)

File: rust/medium/2145_count_the_hidden_sequences.rs
impl Solution {
    pub fn number_of_arrays(differences: Vec<i32>, lower: i32, upper: i32) -> i32 {
        let mut current: i64 = 0;
        let (mut min, mut max) = (0, 0);
        
        for &diff in &differences {
            current += diff as i64;
            min = min.min(current);
            max = max.max(current);
        }
        
        let min_start = lower as i64 - min;
        let max_start = upper as i64 - max;
        
        let result = std::cmp::max(0, max_start - min_start + 1);
        
        if result > i32::MAX as i64 {
            return 0;
        }
        
        result as i32
    }
}
